stream-transformation-test/stream-transformation-performance-test/src/test/java/uk/gov/justice/framework/tools/transformation/StreamTransformationPerformanceIT.java

```diff
@@ -1,78 +1,104 @@
-        assertThat(transformedEventsCount, is(STREAMS_COUNT * EVENTS_PER_STREAM));
-    }
+        final long beforeEventStreamTransform = databaseUtils.getEventStreamJdbcRepository().findAll().count();
 
+        assertThat("================================================================================================\n" +
+                "Incorrect event stream count please check data!"
+                + "================================================================================================\n", beforeEventStreamTransform, is(databaseEventStreamCount));
 
-    private void insertPerformanceTestData() throws InvalidPositionException {
+        final long beforeEventsTransform = databaseUtils.getEventLogJdbcRepository().findAll().count();
+        assertThat("================================================================================================\n" +
+                "Incorrect event log count please check data!", beforeEventsTransform, is(databaseEventLogCount));
 
-        final List<UUID> streamIds = new ArrayList<>();
+        LOGGER.info("Starting performance test");
 
-        for (int i = 0; i < STREAMS_COUNT; i++) {
-            streamIds.add(randomUUID());
-        }
+        final StopWatch stopWatch = new StopWatch();
+        stopWatch.start();
+        LOGGER.info("Stream Count: " + databaseEventStreamCount);
+        LOGGER.info("Total events available for transformation: " + databaseEventLogCount);
+
+        final boolean debug = Boolean.valueOf(getProperty("debug", ENABLE_REMOTE_DEBUGGING_FOR_WILDFLY));
+        final String memoryOptions = getProperty("memoryOptions", DEFAULT_MEMORY);
+
+        final String streamsProcessedCountStepInfo = getProperty("streamsProcessedCountStepInfo", "100");
+        swarmStarterUtil.runCommand(Long.valueOf(streamsProcessedCountStepInfo), debug, WILDFLY_TIMEOUT_IN_SECONDS, memoryOptions);
 
-        final ZonedDateTime createdAt = new UtcClock().now().minusMonths(1);
+        stopWatch.stop();
+        LOGGER.info("Time taken in mill: " + stopWatch.getTime() / 1000);
+        LOGGER.info("================================================================================================");
+        LOGGER.info("Ending performance test");
+        LOGGER.info("================================================================================================");
 
-        for (final UUID id : streamIds) {
-            for (long logId = 1; logId <= EVENTS_PER_STREAM; logId++) {
-                databaseUtils.getEventLogJdbcRepository().insert(eventLogFrom("sample.events.name", logId, id, createdAt));
+    }
+
+    private void generateDataIfRequired() throws InvalidPositionException, SQLException, LiquibaseException {
+        final boolean generateTestData = valueOf(getProperty("generateTestData", GENERATE_TEST_DATA));
+        final long eventsStreamCount = Long.valueOf(getProperty("eventsStreamCount"));
```

Comment:
can't we only extract these additional values within the IF condition below

Reply:
fixed